### DISCUSSION TITLE
NextAuth environment variables

### DISCUSSION THREAD

**Discussion by AGeck113** [neuefische/questions#526] [Web] [answered]
Hello together, 
I tried to implement next Auth in my project. 
It works fine on my localhost, but not in the preview. 
I created the OAuth on Github or the Preview and implemented the GITHUB_ID and the GITHUB_SECRET, and also my NextAuth Secret for the preview Deployment.
If I open the page, i see that in my console: 

<img width="453" alt="image" src="https://user-images.githubusercontent.com/117731514/218120194-465b2286-1f67-465c-9a24-176b8d681afb.png">

So the problem should be in the env. variables but I checked them at least 3 times. 

I can also see in my Github account, that the Secret was used 0 times. So I think it doesn't make it to that point, and therefore, I would assume that the Problem is at the NEXTAUTH_SECRET, but i tried it with the same one as local and with a completely new one and no of the 2 Secrets worked.... I searched for errors but can't find anything.

**Comment by doemser** [comment, marked as answer]
Two things to check:
- the server logs - your preview has its own server logs on vercel
- your oAuth App needs to use the url of your preview, for Callback and Authorization Callback

**Reply by doemser** [reply]
Also make sure to redeploy if you change something to .env variables, you can do that via vercel as well

**Reply by AGeck113** [reply]
Thanks, a simple redeploy was the answer. Checked my envs multiple times but didn't think of that 😅
